integrationtests/tests/tests.rs

```diff
@@ -782,3 +802,50 @@ async fn rejoin_consensus_threshold_peers() {
     // confirm that the entire federation can rejoin at an epoch
     timeout(Duration::from_secs(15), rejoin).await.unwrap();
 }
+
+#[tokio::test(flavor = "multi_thread")]
+async fn test_ln_client_adapter() {
```

Comment:
Sounds to me like a unit test, where we validate the adapter behavior, before we apply it in our integration testing

Reply:
yeah so this is basically testing part of the "testing api" which does not make alot of sense but I wanted to see if this works. I'll follow up with a gateway retry for failed payments PR which is going to use this anyway so we can remove this test here.

Reply:
> Sounds to me like a unit test, where we validate the adapter behavior, before we apply it in our integration testing

I don't have a lot of experience with stuff like that, do you think this should get a unit test, if yes where to put that (in `mod` ?)

Reply:
> basically testing part of the "testing api"

Generally we can just rely on the integration tests themselves to "test" the test framework.  Perhaps we can eliminate this test and just cherry-pick the test that @okjodom wrote on top of this commit.

Reply:
yeah that was the plan, I was not aware @okjodom s test was already in

Reply:
Oh no, I have not written it on top of this commit. But could. Here is the reference test, that I'm hoping is easier to do after our reworking of fixtures: https://github.com/fedimint/fedimint/pull/602/files#r979193260

Reply:
> Generally we can just rely on the integration tests themselves to "test" the test framework

Agreed. The testing framework self validates as we use it to write actual tests. Especially in this case, we don't need to specially test the adapter because we'll know it works if it makes the LnRpc adopt the right behavior we require in our test scenarios.

Now if we had to write a test for the test framework, that should be a unit test. Right?

Reply:
I guess a unit test is unnecessary then. should I remove this test or put a `TODO: First one that writes a test that uses LnRpcAdapter shall remove this test` on it ?